launchpod/angular-launchpod/backend/src/main/java/com/google/launchpod/servlets/FormHandlerServlet.java

```diff
@@ -0,0 +1,301 @@
+  public static final String HTML_FORM_END = "  <input type='file' name='file'/><br />\n"
+                                             + "<input type='submit' value='Upload File' name='submit'/><br />\n"
+                                             + "</form>\n";
+
+  private static final String ID = "id";
+  private static final String ACTION = "action";
+
+  private enum Action {
+    generateRSSLink("generateRSSLink"),
+    generateXml("generateXml"),
+    otherAction("otherAction"); // for testing purposes
+
+    private String action;
+ 
+    Action(String action) {
+        this.action = action;
+    }
+
+    @Override
+    public String toString() {
+      return action;
+    }
+  }
+
+  /**
+  * Creates an MP3 link from GCS bucket name and the object name (entity id).
+  */
+  private String makeMp3Link(String entityId) {
+    String link = "https://storage.googleapis.com/" + BUCKET_NAME + "/" + entityId;
+    return link;
+  }
+
+  /**
+   * Requests user inputs in form fields, then creates Entity and places in Datastore.
+   * @throws ServletException
+   */
+  @Override
+  public void doPost(HttpServletRequest req, HttpServletResponse res) throws IllegalArgumentException, IOException {
+    String podcastTitle = req.getParameter(PODCAST_TITLE);
+    String description = req.getParameter(DESCRIPTION);
+    String language = req.getParameter(LANGUAGE);
+    String email = req.getParameter(EMAIL);
+
+    // TO-DO after merging: add this in and remove email parameter (because LoginServlet will be set up)
+    // UserService userService = UserServiceFactory.getUserService();
+    // if (userService.isUserLoggedIn()) {
+    //   email = userService.getCurrentUser().getEmail();
+    // }
+
+    // if (podcastTitle == null || podcastTitle.isEmpty()) {
+    //   throw new IllegalArgumentException("No Title inputted, please try again.");
+    // } else if (description == null || description.isEmpty()) {
+    //   throw new IllegalArgumentException("No description inputted, please try again.");
+    // } else if (language == null || language.isEmpty()) {
+    //   throw new IllegalArgumentException("No language inputted, please try again.");
+    // } else if (email == null || email.isEmpty()) {
+    //   throw new IllegalArgumentException("You are not logged in. Please try again.");
+    // }
+
+    if (Strings.isNullOrEmpty(podcastTitle)) {
+      throw new IllegalArgumentException("No Title inputted, please try again.");
+    } else if (Strings.isNullOrEmpty(description)) {
+      throw new IllegalArgumentException("No description inputted, please try again.");
+    } else if (Strings.isNullOrEmpty(language)) {
+      throw new IllegalArgumentException("No language inputted, please try again.");
+    } else if (Strings.isNullOrEmpty(email)) {
+      throw new IllegalArgumentException("You are not logged in. Please try again.");
+    }
```

Comment:
this is copied code - there should be a common place that performs this validation

Reply:
across FormHandlerServlet and FileUploadServlet?

Reply:
will be done in a separate PR specifically meant for common methods and constants